internal/controller/clusterinstance_controller.go

```diff
@@ -356,22 +373,65 @@ func createOrPatch(
 			}
 		}
 
-		if err := c.Create(ctx, &obj); err != nil {
+		if err := c.Create(ctx, &renderedObj); err != nil {
 			return controllerutil.OperationResultNone, err
 		}
 		return controllerutil.OperationResultCreated, nil
 	}
 
 	// Object exists, update it
-	obj.SetResourceVersion(existingObj.GetResourceVersion())
-	obj.SetOwnerReferences(existingObj.GetOwnerReferences())
-	patch := client.MergeFrom(existingObj)
+	patch := client.MergeFrom(liveObj.DeepCopy())
+	updatedLiveObj := liveObj.DeepCopy()
+
+	// If the resource contains a status then remove it from the unstructured
+	// copy to avoid unnecessary patching later.
+	unstructured.RemoveNestedField(updatedLiveObj.Object, "status")
+
+	// Merge metadata.annotations and metadata.labels
+	updatedLiveObj.SetAnnotations(mergeMaps(renderedObj.GetAnnotations(), liveObj.GetAnnotations()))
+	updatedLiveObj.SetLabels(mergeMaps(renderedObj.GetLabels(), liveObj.GetLabels()))
+
+	// Replace updatedLiveObj.spec fields with those in renderedObj.spec
+	spec, ok := renderedObj.Object["spec"].(map[string]interface{})
+	if !ok {
+		return controllerutil.OperationResultNone, fmt.Errorf("missing spec in rendered install template")
+	}
+	for _, key := range maps.Keys(spec) {
+		nestedField := []string{"spec", key}
+		nestedValue, ok, err := unstructured.NestedFieldCopy(renderedObj.Object, nestedField...)
+		if err != nil {
+			return controllerutil.OperationResultNone, err
+		}
+		if !ok {
```

Comment:
Maybe add a comment here about how we should really never be in this situation because this would mean that a field in the keys for the rendered object spec is now not in the rendered object spec somehow.

Reply:
Done!